packages/neon_framework/lib/src/pages/account_settings.dart

```diff
@@ -45,32 +46,37 @@ class AccountSettingsPage extends StatelessWidget {
       actions: [
         IconButton(
           onPressed: () async {
-            final decision = await showAdaptiveDialog<bool>(
+            final decision = await showAdaptiveDialog<AccountDeletion>(
               context: context,
-              builder: (context) => NeonConfirmationDialog(
-                icon: const Icon(Icons.logout),
-                title: NeonLocalizations.of(context).accountOptionsRemove,
-                content: Text(
-                  NeonLocalizations.of(context).accountOptionsRemoveConfirm(account.humanReadableID),
-                ),
+              builder: (context) => NeonAccountDeletionDialog(
+                account: account,
               ),
             );
 
-            if (decision ?? false) {
-              final isActive = bloc.activeAccount.valueOrNull == account;
+            switch (decision) {
+              case null:
+                break;
+              case AccountDeletion.remote:
+                await launchUrl(
+                  account.serverURL.replace(
+                    path: '${account.serverURL.path}/index.php/settings/user/drop_account',
+                  ),
+                );
+              case AccountDeletion.local:
+                final isActive = bloc.activeAccount.valueOrNull == account;
 
-              options.reset();
-              bloc.removeAccount(account);
+                options.reset();
+                bloc.removeAccount(account);
 
-              if (!context.mounted) {
-                return;
-              }
+                if (!context.mounted) {
+                  return;
+                }
 
-              if (isActive) {
-                const HomeRoute().go(context);
-              } else {
-                Navigator.of(context).pop();
-              }
+                if (isActive) {
+                  const HomeRoute().go(context);
+                } else {
+                  Navigator.of(context).pop();
+                }
```

Comment:
Does this mean the local account is not deleted when the user selects a remote deletion?

Reply:
Yes, because the user could still cancel the account deletion in the browser.
We could change the dialog to only have a checkbox (in case the app is available) and the user would be logged out in any case, but taken to the web page after being logged out. I'm not sure if that is a better UX, but it would be a viable alternative.

Reply:
Once we support in app deletion requests we should also use log the user out.
I think it might also be expected now but I'll leave that decision to you.

Reply:
Ok then I'll keep it this way for now and once we can directly delete the user from within the app we should log them out as well.